### ISSUE TITLE
Question on pose space training

### ISSUE THREAD

**Issue by eddiecong** [Hangz-nju-cuhk/Talking-Face_PC-AVS#21]
Thanks so much for the great work and codes. When I read the paper and codes, I get confused about the pose space learning part.

As the training strategy said in the paper, firstly pre-train the identity encoder and speech content space and then loaded to the overall framework to train the generator and pose space learning, I could understand the training procedure, however, for learning the pose space I am confused **if you use the loss (compute_diff_loss)** 

https://github.com/Hangz-nju-cuhk/Talking-Face_PC-AVS/blob/23585e281360872fe2d1e1eec8ff49176ea0183d/models/av_model.py#L401

when training the whole generator. If so, it is consistent with the codes to compute the l1 loss on pose differences and pose_feature_audio. 

Last but not least, congratulations on the research progress, I think it is a great breakthrough to disentangle the sync information and head pose in the feature representation. Looking forward to your reply!

**Comment by Hangz-nju-cuhk** [comment]
Hi, this loss is not used in this paper (PC-AVS).

**Comment by eddiecong** [comment]
@Hangz-nju-cuhk Thanks so much for the reply. Could I summary the training procedure into these three stages?

1. pre-train the identity encoder.
2. pre-train the speech content space (audio encoder, video encoder, and mouth module).
3. train the generator, together with the pose space module, at the same time freezing the video encoder, audio encoder, and identity encoder.

Also, for the pose module, not using the supervised learning of 12 parameters, only let the model unsupervised learn the representation of pose.

Thanks in advance, this will extremely help my implementation of the training step.

**Comment by Hangz-nju-cuhk** [comment]
Hi, the pipeline is basically correct. I have found out that between 1 and 2, if the non-identity space is pretrained in a visual-driven manner, the pretrained weights can help the learning of the speech content space and the generator.

**Comment by eddiecong** [comment]
Thanks for the reply, that's really helpful!
